src/Microsoft.TemplateEngine.Utils/InstallRequestPathResolution.cs

```diff
@@ -26,53 +28,73 @@ public static class InstallRequestPathResolution
             {
-                yield return maskedPath;
-                yield break;
+                throw new ArgumentException($"'{nameof(maskedPath)}' cannot be null or whitespace.", nameof(maskedPath));
+            }
+
+            if (environmentSettings is null)
+            {
+                throw new ArgumentNullException(nameof(environmentSettings));
             }
-            var matches = DetermineDirectoriesToScan(maskedPath, environmentSettings).ToList();
-            //This can happen when user specifies "PackageId"
-            if (matches.Count == 0)
+
+            var arr = Path.GetInvalidPathChars();
+            if (maskedPath.IndexOfAny(arr) != -1)
             {
                 yield return maskedPath;
                 yield break;
             }
-            foreach (var path in matches)
+
+            foreach (string path in ResolveSearchPattern(maskedPath, environmentSettings))
             {
-                yield return Path.GetFullPath(path);
+                yield return path;
             }
         }
 
-        private static IReadOnlyList<string> DetermineDirectoriesToScan(string baseDir, IEngineEnvironmentSettings environmentSettings)
+        private static IEnumerable<string> ResolveSearchPattern(string maskedPath, IEngineEnvironmentSettings environmentSettings)
         {
-            List<string> directoriesToScan = new List<string>();
+            string baseDir = maskedPath;
 
+            //trim trailing separators
             if (baseDir[baseDir.Length - 1] == '/' || baseDir[baseDir.Length - 1] == '\\')
             {
                 baseDir = baseDir.Substring(0, baseDir.Length - 1);
             }
 
-            string searchTarget = Path.Combine(environmentSettings.Host.FileSystem.GetCurrentDirectory(), baseDir.Trim());
-            List<string> matches = environmentSettings.Host.FileSystem.EnumerateFileSystemEntries(Path.GetDirectoryName(searchTarget), Path.GetFileName(searchTarget), SearchOption.TopDirectoryOnly).ToList();
-
-            if (matches.Count == 1)
+            try
             {
-                directoriesToScan.Add(matches[0]);
-            }
-            else
-            {
-                foreach (string match in matches)
+                string searchTarget = Path.Combine(environmentSettings.Host.FileSystem.GetCurrentDirectory(), baseDir.Trim());
+                string parentFolder = Path.GetFullPath(Path.GetDirectoryName(searchTarget));
+                string searchPattern = Path.GetFileName(searchTarget);
+
+                //EnumerateFileSystemEntries treats '.' as '*' and cannot handle '..'
+                if (searchPattern.Equals(".") || searchPattern.Equals(".."))
+                {
+                    string normalizedPath = Path.GetFullPath(searchTarget);
+                    if (environmentSettings.Host.FileSystem.FileExists(normalizedPath) || environmentSettings.Host.FileSystem.DirectoryExists(normalizedPath))
```

Comment:
nit: Do we really need to check for file here?
Would it make more sense to just do `if (".") return Directory.GetCurrentDirectory();` and `if ("..") return Path.GetDirectory(Directory.GetCurrentDirectory());`?

Reply:
We checked for existence before via (EnumerateFileSystemEntries), but probably it will always exist except `c:\..` case.